### PR TITLE
Update YamlParser and YamlPrinter to support Mapping and Sequence anchors. (issue #1161)

### LINKED ISSUE #1161: Anchors disappear in YamlParser for Scalar keys, Sequences, and Mapping nodes.
`Anchor` and `Alias` have been added to the Yaml model, but only work on Scalar values.
Anchors on Scalar keys, Sequences, and Mapping nodes will disappear.
Aliases for any of the mentioned Anchors will NPE.

Notes:
- Based on the Yaml spec, each Yaml#Node includes an Anchor.
- Scalar, Sequences, Mapping, and Alias are Yaml#Nodes.
- Our equivalent to Yaml#Node is Yaml#Block, since Scalar, Sequence, Mapping, and Alias implement Block.
- Yaml `MappingNode` is a key: `Node` =>  value: `Node`, but our `Mapping#Entry` is a key: `Yaml#Scalar` => value: `Yaml#Block`.

### PR DESCRIPTION
**Pull request by pway99**

### CODE REVIEW

File: rewrite-yaml/src/test/kotlin/org/openrewrite/yaml/tree/MappingTest.kt [hunk 1]
```diff
@@ -229,13 +229,25 @@ class MappingTest: YamlParserTest {
...
+    @Test
+    fun scalarEntryValue() = assertRoundTrip("""
+         foo:
+          - start: start
+          - buz: &anchor ooo
+          - fuz: *anchor
+          - end: end
+    """)
+
     @Disabled
```
Comment (nmck257) [inline rewrite-yaml/src/test/kotlin/org/openrewrite/yaml/tree/MappingTest.kt:241]: Confirming scope:
* This case (and `scalarKeyAnchor` up at line 224) are still disabled. relatedly, And I believe YAML allows anchors on mapping keys, too. Are those included in this PR?
* I see the `mappingAnchor` case below is no longer disabled, and it features the merge-key syntax `<<`. So, this PR covers https://github.com/openrewrite/rewrite/issues/2133, too?

Comment (pway99) [reply rewrite-yaml/src/test/kotlin/org/openrewrite/yaml/tree/MappingTest.kt:241]: >This case (and scalarKeyAnchor up at line 224) are still disabled. relatedly, And I believe YAML allows anchors on mapping keys, too. Are those included in this PR?

Thanks for noticing, yes I would agree the YAML does allow anchors on mapping keys also.
https://yaml.org/spec/1.2.2/#3222-anchors-and-aliases

It's interesting though while the spec seems to allow defining an anchor for a key I have not been able to find a YAML validator that successfully validates the example. 

Sequence entry validation error:
```YAML
foo:
  - start: start
  - &anchor buz: buz ## Bad indentation for a sequence entry validation error
  - *anchor: baz
  - end: end
```
An example that passes validation:
```YAML
bar:
  &abc yo: friend
baz:
  *abc: friendly
```

I will give this a bit more investigation and see if I can't make it work for keys also

>I see the mappingAnchor case below is no longer disabled, and it features the merge-key syntax <<. So, this PR covers https://github.com/openrewrite/rewrite/issues/2133, too?

Good question I will have a look and get back to you on that also.

Comment (pway99) [reply rewrite-yaml/src/test/kotlin/org/openrewrite/yaml/tree/MappingTest.kt:241]: @nmck257 I am going to merge these changes and leave the issue open then work on the keys having anchors and aliases in another PR

Comment (traceyyoshima) [reply rewrite-yaml/src/test/kotlin/org/openrewrite/yaml/tree/MappingTest.kt:241]: YamlLinter: [link](http://www.yamllint.com/)

**Pull request merged**